src/sentry/issues/occurrence_status_change.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from typing import Sequence
+
+from sentry.models.group import Group, GroupStatus
+from sentry.types.activity import ActivityType
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class StatusChangeOccurrence:
```

Comment:
I'm not totally sure we should have occurrence in the name here. An occurrence is "an occurrence of an issue", basically an event. But we're changing the status of an issue, so I'm wondering if something like
`PlatformStatusChange`, `IssueStatusChange` or `StatusChangeMessage` make more sense.